src/publicApi.ts

```diff
@@ -0,0 +1,48 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Gitpod. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { createConnectTransport, createPromiseClient, Interceptor, PromiseClient } from '@bufbuild/connect-web';
+import { WorkspacesService } from '@gitpod/public-api/lib/gitpod/v1/workspaces_connectweb';
+import { Workspace, WorkspaceInstance } from '@gitpod/public-api/lib/gitpod/v1/workspaces_pb';
+
+export class GitpodPublicApi {
+
+    private workspaceService!: PromiseClient<typeof WorkspacesService>;
```

Comment:
Why not make it public and put all needed services here with init or cache (cache with a getter?)